### ISSUE TITLE
SPIRV 1.3 support

### ISSUE THREAD

**Issue by aschrein** [vulkano-rs/vulkano#1211]
*   Version of vulkano: 0.11
*   OS: Arch Linux

# Issue

Hello,

I want to use [subgroup shuffle extension](https://www.khronos.org/blog/vulkan-subgroup-tutorial)
I'm getting 'Error: subgroup op: requires SPIR-V 1.3' when compiling with vulkano_shaders::shader! macro.

I believe the problem stems from the default vulkan 1.0 target, changing it to vulkan 1.1 might resolve the issue. CompileOptions trait has [set_target_env](https://github.com/google/shaderc-rs/blob/d62f6720991397665c09c16a59440da5e345ee7c/shaderc-rs/src/lib.rs#L678) method which, I think, can be used to set the right target before invoking compiler.compile_into_spirv [here](https://github.com/vulkano-rs/vulkano/blob/d98552fe41a9e7c7df1e87c93cc06ea9d3fd81b5/vulkano-shaders/src/codegen.rs#L150). However, I'm not sure if changing the target in that way won't break something else.

Using runtime shaders is a workaround.

Thanks

**Comment by AustinJ235** [comment]
Is this still an issue?

**Comment by aschrein** [comment]
Not sure. Haven't played with vulkano since.
Having skimmed through the latest commit messages I can't say that default spirv version has been changed. Shaderc has been updated though

**Comment by AustinJ235** [comment]
Investigating another issue I stumbled upon some code to do with instance creation. In there we do specify vulkan 1.0 to be used. We can certainly look into bumping this or providing the user a way to chose their own version.

**Comment by AustinJ235** [comment]
This is where the version is defined if you wanted to have a go: https://github.com/vulkano-rs/vulkano/blob/761fb4e6338b86926bd25c30d382aa98e2b28e5c/vulkano/src/instance/instance.rs#L223

**Comment by AustinJ235** [comment]
I didn't experience any issues upgrading to vulkan 1.1 within vulkano.

https://github.com/vulkano-rs/vulkano/pull/1258 bumps the version to 1.1 and tells shaderc to use target vulkan 1.1 this defaults to spirv version 1.3 however. I already have some code to allow customizing the spirv version within the `shader!` macro, but am currently blocked by the fact that `shaderc-rs` doesn't export the function `shaderc_compile_options_set_target_spirv`, https://github.com/vulkano-rs/vulkano/issues/1259.
